extensions/BeatBlox/index.js

```diff
@@ -627,12 +628,25 @@
                         this.runAsyncFn(async () => {
                             await instrumentPrefetch; // wait for all instruments to be loaded
                             const trackName = this.receiver.id;
-                            const t = await playClip(trackName, clip, this.musicInfo.t, duration);
-                            this.musicInfo.t += t;
-                            await waitUntil(this.musicInfo.t - SCHEDULING_WINDOW);
+                            const clipDuration = await getAudioObjectDuration(clip.audio);
+                            if(duration > clipDuration) {
+                                let remainingDuration = duration;
+                                while (remainingDuration > 0){
+                                    const playingDuration = Math.min(remainingDuration, clipDuration);
+                                    const t = await playClip(trackName, clip, this.musicInfo.t, playingDuration);
+                                    this.musicInfo.t += t;
+                                    await waitUntil(this.musicInfo.t - SCHEDULING_WINDOW);
+                                    remainingDuration -= playingDuration;
+                                } 
+                            }
+                            else{
+                                const t = await playClip(trackName, clip, this.musicInfo.t, duration);
+                                this.musicInfo.t += t;
+                                await waitUntil(this.musicInfo.t - SCHEDULING_WINDOW);
+                            }
                         }, { args: [], timeout: I32_MAX });
                     }),
-                    new Extension.Block('playSampleForDuration', 'command', 'music', 'play sound %snd for duration %noteDurations %noteDurationsSpecial', [null, 'Quarter', ''], function (clip, duration, durationSpecial) {
+                    new Extension.Block('playSampleForDuration', 'command', 'music', 'play clip %snd for duration %noteDurations %noteDurationsSpecial', [null, 'Quarter', ''], function (clip, duration, durationSpecial) {
```

Comment:
This should probably just convert the beat duration to seconds and call the other function. No need to duplicate all the logic.